src/main/java/seedu/address/logic/parser/EditBuyerCommandParser.java

```diff
@@ -0,0 +1,91 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_BUDGET;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_HOUSING_TYPE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.EditBuyerCommand;
+import seedu.address.logic.commands.EditBuyerCommand.EditBuyerDescriptor;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new EditBuyerCommand object
+ */
+public class EditBuyerCommandParser implements Parser<EditBuyerCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the EditBuyerCommand
+     * and returns an EditBuyerCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public EditBuyerCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_HOUSING_TYPE,
+                        PREFIX_BUDGET, PREFIX_TAG);
+
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditBuyerCommand.MESSAGE_USAGE), pe);
+        }
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_HOUSING_TYPE,
+                PREFIX_BUDGET);
+
+        EditBuyerDescriptor editBuyerDescriptor = new EditBuyerDescriptor();
+
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            editBuyerDescriptor.setName(ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get()));
```

Comment:
Any way to compress the if statements or abstract them away? If not looks okay

Reply:
Yeahh I guess the chaining is too long.

Would it be better if I split it this way?
```
        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
        
        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
            editBuyerDescriptor.setName(name);
        }
```